public/transaction-list.js

```diff
@@ -0,0 +1,367 @@
+async function buildJobsTable(transactionsTable, transTableHeader, token, message) {
+    try {
+        const response = await fetch("/api/v1/transactions", {
+            method: "GET",
+            headers: {
+                "Content-Type": "application/json",
+                Authorization: `Bearer ${token}`,
+            },
+        });
+        const data = await response.json();
+        var children = [transTableHeader];
+        if (response.status === 200) {
+            if (data.count === 0) {
+                transactionsTable.replaceChildren(...children); // clear this for safety
+                return 0;
+            } else {
+                for (let i = 0; i < data.transactions.length; i++) {
+                    let editButton = `<td><button type="button" class="editButton" data-id=${data.transactions[i]._id}>edit</button></td>`;
+                    let deleteButton = `<td><button type="button" class="deleteButton" data-id=${data.transactions[i]._id}>delete</button></td>`;
+
+                    let rowHTML = ` <td>${data.transactions[i].transType}</td>
+                                    <td>${data.transactions[i].category}</td>
+                                    <td>${data.transactions[i].amount}</td>
+                                    ${editButton}
+                                    ${deleteButton}
+                                    `;
+                    
+                    let rowEntry = document.createElement("tr");
+                    rowEntry.innerHTML = rowHTML;
+                    children.push(rowEntry);
+                }
+                transactionsTable.replaceChildren(...children);
+            }
+            return data.count;
+        } else {
+            message.textContent = data.msg;
+            return 0;
+        }
+    } catch (err) {
+        message.textContent = "A communication error occurred.";
+        return 0;
```

Comment:
Here's a bonus challenge -- it can be good to have a `/logger` endpoint that will log client-side errors from all clients to the backend, or even save them to the database, optionally. Otherwise, you have no visibility when something goes wrong on the client in production!